arroyo/processing/strategies/dead_letter_queue/policies/produce.py

```diff
@@ -0,0 +1,89 @@
+import base64
+import json
+import time
+from collections import deque
+from concurrent.futures import Future
+from typing import Deque, Optional
+
+from arroyo.backends.kafka.consumer import KafkaPayload, KafkaProducer
+from arroyo.processing.strategies.dead_letter_queue.policies.abstract import (
+    DeadLetterQueuePolicy,
+    InvalidMessages,
+    Serializable,
+)
+from arroyo.types import Message, Topic
+from arroyo.utils.metrics import get_metrics
+
+MAX_QUEUE_SIZE = 5000
+
+
+class ProduceInvalidMessagePolicy(DeadLetterQueuePolicy):
```

Comment:
Please have a unit test for this step. Please rely on this producer https://github.com/getsentry/arroyo/blob/main/arroyo/backends/local/backend.py#L356-L398. so you do not need to bring in all the Kafka complexity in your test. Then you can check all the messages that you produced from the Broker object.

Reply:
Added a simple test for one good and one bad message, should it be more extensive?